test/arrayops.jl

```diff
@@ -2900,29 +2900,29 @@ end
     b = IOBuffer()
     showerror(b, err)
     @test String(take!(b)) ==
-        "BoundsError: attempt to access 2×2 Matrix{Float64} at index [10, 1:2]"
+        "BoundsError: attempt to access 2×2 Matrix{Float64} at index [10, 1:2].\nLegal indices are [1:2, 1:2]."
 
     err = try x[10, trues(2)]; catch err; err; end
     b = IOBuffer()
     showerror(b, err)
     @test String(take!(b)) ==
-        "BoundsError: attempt to access 2×2 Matrix{Float64} at index [10, 2-element BitVector]"
+        "BoundsError: attempt to access 2×2 Matrix{Float64} at index [10, 2-element BitVector].\nLegal indices are [1:2, 1:2]."
 
     # Also test : directly for custom types for which it may appear as-is
     err = BoundsError(x, (10, :))
     showerror(b, err)
     @test String(take!(b)) ==
-        "BoundsError: attempt to access 2×2 Matrix{Float64} at index [10, :]"
+        "BoundsError: attempt to access 2×2 Matrix{Float64} at index [10, :].\nLegal indices are [1:3, 1:3]."
```

Comment:
@gustaphe Thank you for your contribution and welcome to the community. This does make error messege clearer!
Beside the the above typo. I guess you also need to fix the test in `Test` and resolve the mismatches in doc examples.

Reply:
Sorry, I have a hard time running the tests, they end up running my laptop's memory into the ground and causing a reboot. I thought I found a subset of the tests that would catch everything but sadly some slipped through. I don't want it to seem like I'm brute force debugging this by spamming commits, but I don't have any other method of knowing it works so I kind of am.

Reply:
It is okay to brute force it with CI time. We usually have a little bit of excess. This seems like a good idea. It might conflict eventually with https://github.com/JuliaLang/julia/pull/43738, but let's cross the bridge later, and hopefully merge this sooner.